sklearn/preprocessing/tests/test_target_encoder.py

```diff
@@ -701,3 +701,10 @@ def test_target_encoding_for_linear_regression(smooth, global_random_seed):
     # cardinality yet non-informative feature instead of the lower
     # cardinality yet informative feature:
     assert abs(coef[0]) < abs(coef[2])
+
+
+def test_27879():
+    pd = pytest.importorskip("pandas")
```

Comment:
For the [failing CI](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=63388&view=logs&j=aabdcdc3-bb64-5414-b357-ed024fe8659e&t=08761b20-68b0-57e8-3399-a73e66658a1f), it is using a really old version of pandas that does not have `mode.copy_on_write`. Here we can set the min version here: 

```suggestion
    pd = pytest.importorskip("pandas", minversion="2.0")
```